app/celery/tasks.py

```diff
@@ -269,7 +269,7 @@ def save_smss(self, service_id: Optional[str], signed_notifications: List[Signed
 
         verified_notifications.append(notification)
         notification_id_queue[notification_id] = notification.get("queue")  # type: ignore
-        process_type = template.process_type
+        process_type = str(template.process_type)
```

Comment:
Kept running into an incompatible type error from MyPy:

```
app/celery/tasks.py:403: error: Argument "priority" to "put_batch_saving_bulk_processed" has incompatible type "Union[Any, MapperProperty[Any], ColumnElement[Any]]"; expected "Optional[str]"
```
In [models.py](https://github.com/cds-snc/notification-api/blob/be4177e240b374e2520cb75bc6957fd26d1925ea/app/models.py#L1067) it appears that `@declared_attr` annotated properties aren't inferred by MyPy without [jumping through hoops](https://docs.sqlalchemy.org/en/14/orm/extensions/mypy.html#using-declared-attr-and-declarative-mixins). It may have something to do with invariance vs covariance as [described in MyPy documentation](https://mypy.readthedocs.io/en/stable/common_issues.html#invariance-vs-covariance). 

`sqlalchemy-stubs` and `sqlalchemy2-stubs` are supposed to provide us with better type inference for SQLAlchemy related types. However enabling the plugin in `mypy.ini` seems to have no effect.

We should revisit some of the typing we do to avoid str() calls just to satisfy MyPy.

Reply:
I think we may need to migrate to sqlalchemy 2.0 to improve how the typing is working - it looks like that adds better support for `Column` objects: https://docs.sqlalchemy.org/en/20/changelog/whatsnew_20.html#new-typing-support-in-core-and-orm-stubs-extensions-no-longer-used

Reply:
Does adding a `# type ignore` comment to that line instead of `str()` fix the error? Then we wouldn't have to change the code.

Reply:
Removing the `str()` and putting `# type: ignore` on L290 and L403 seem to resolve the issue. We could get some type-ception going on here and use the typed ignore `# type: ignore[arg-type]`.